subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java

```diff
@@ -210,30 +208,34 @@ private boolean deleteDir(File dir) {
         return dir.delete();
     }
 
-    private void unzip(File zip, File dest) throws IOException {
-        Enumeration entries;
-        ZipFile zipFile = new ZipFile(zip);
-
+    private void unzip(File zip, File dest) {
         try {
-            entries = zipFile.entries();
-
-            while (entries.hasMoreElements()) {
-                ZipEntry entry = (ZipEntry) entries.nextElement();
-
-                if (entry.isDirectory()) {
-                    (new File(dest, entry.getName())).mkdirs();
-                    continue;
-                }
-
-                OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(new File(dest, entry.getName())));
-                try {
-                    copyInputStream(zipFile.getInputStream(entry), outputStream);
-                } finally {
-                    outputStream.close();
+            Enumeration entries;
+            ZipFile zipFile = new ZipFile(zip);
+            try {
+                entries = zipFile.entries();
+
+                while (entries.hasMoreElements()) {
+                    ZipEntry entry = (ZipEntry) entries.nextElement();
+
+                    if (entry.isDirectory()) {
+                        (new File(dest, entry.getName())).mkdirs();
+                        continue;
+                    }
+
+                    OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(new File(dest, entry.getName())));
+                    try {
+                        copyInputStream(zipFile.getInputStream(entry), outputStream);
+                    } finally {
+                        outputStream.close();
+                    }
                 }
+            } finally {
+                zipFile.close();
             }
-        } finally {
-            zipFile.close();
+        } catch (IOException e) {
+            logger.log("Could not unzip " + zip.getAbsolutePath() + " to " + dest.getAbsolutePath() + ".");
+            logger.log("Reason: " + e.getMessage());
```

Comment:
:x: I guess users potentially would like to see the stack trace. I'd probably go with `logger.log("Reason: ", e);`.

Reply:
Hmm. I can change it because hopefully very few folks will see it, but I feel like the vast majority of issues are captured well by the message "invalid zip" or "permissions somethingorother". A stacktrace will sure get folks attention, though.

Reply:
The stacktrace is printed by an upstream method when this exception is rethrown

Reply:
ok